src/main/java/com/warrenstrange/googleauth/GoogleAuthenticator.java

```diff
@@ -166,10 +168,56 @@ public GoogleAuthenticator(GoogleAuthenticatorConfig config)
         }
 
         this.config = config;
+
+        this.secureRandom = new ReseedingSecureRandom(
+                getRandomNumberAlgorithm(),
+                getRandomNumberAlgorithmProvider()
+        );
+    }
+
+    public GoogleAuthenticator(final String randomNumberAlgorithm, final String randomNumberAlgorithmProvider)
+    {
+        config = new GoogleAuthenticatorConfig();
```

Comment:
Should this constructor chain to the one below to avoid duplicating the preconditions?

Reply:
Good idea https://github.com/wstrange/GoogleAuth/pull/87/commits/0459feb0190bb7e9d4ee6959a2c9a10f5e9963c0